modules/entity/src/unsorted_state_adapter.ts

```diff
@@ -112,13 +112,41 @@ export function createUnsortedStateAdapter<T>(selectId: IdSelector<T>): any {
     return didMutate;
   }
 
+  function upsertOneMutably(update: Update<T>, state: R): boolean;
+  function upsertOneMutably(update: any, state: any): boolean {
+    return upsertManyMutably([update], state);
+  }
+
+  function upsertManyMutably(updates: Update<T>[], state: R): boolean;
+  function upsertManyMutably(updates: any[], state: any): boolean {
+    const added: T[] = [];
+    const updated: Update<T>[] = [];
+
+    for (let index in updates) {
+      const update = updates[index];
+      if (update.id in state.entities) {
+        updated.push(update);
+      } else {
+        added.push({
+          ...update.changes,
+          id: update.id,
```

Comment:
This line is problematic: if `selectId != entity => entity.id`, this insert will fail to create an entity with the same effective id.

Reply:
Let's make this change.